app/controllers/application_controller.rb

```diff
@@ -77,18 +77,18 @@ def check_api_version
 
   def check_client_version(client_name, client_version)
     begin
-      client_version = Version.new(client_version) if client_version != nil
+      client_version = Version.new(client_version) unless client_version.nil?
     rescue
       raise "Invalid version string: #{client_version}"
     end
 
     valid_clients = SiteSetting.value('valid_clients')
     if valid_clients.is_a?(Enumerable)
-      vc = valid_clients.find {|c| c['name'] == client_name }
-      raise "Invalid TMC client." if vc == nil
+      vc = valid_clients.find { |c| c['name'] == client_name }
+      fail 'Invalid TMC client.' if vc.nil?
```

Comment:
https://github.com/bbatsov/ruby-style-guide#exceptions. On line 82, RuboCop probably considers this as re-raising. Maybe `fail` would make more sense in this case.